### PR TITLE
WIP: Added named styles

### PR DESCRIPTION
**Pull request by vinothvino42**

### CODE REVIEW

File: lib/widget/button.dart [hunk 1]
```diff
@@ -33,6 +37,15 @@ class Button extends StatefulWidget
 
   @override
   Map<String, Function> setters() {
+    final styleModel = GetIt.I<StyleProvider>();
+    final StyleTheme? style = styleModel.getStyleName(_controller.styleName);
+
+    if (style != null) {
+      _controller.color = _controller.color ?? style.color;
```
Comment (vusters) [inline lib/widget/button.dart:44]: This doesn't scale. None of the widgets should have to know about styleName.

Comment (vinothvino42) [reply lib/widget/button.dart:44]: This kind of style will be applied to a widget only if it has the `styleName` right? So each widget is supposed to know its styleName right? If it has the styleName, then the style with that name is applied to that widget.

Comment (vusters) [reply lib/widget/button.dart:44]: @amin-nas i thought we said styleName has no type, and can be applied to any widget.

Comment (amin-nas) [reply lib/widget/button.dart]: Correct. I believe the ticket here was updated: https://github.com/EnsembleUI/ensemble/issues/432

Comment (vinothvino42) [reply lib/widget/button.dart]: @vusters Okay no need to add type for the widget in styleName but we need to store the styleName for each widget. If it has the styleName, then the style with that name is applied to that widget.

**Comment by kmahmood74** [comment]
@vinothvino42 is this complete? if yes, always close after merging please.

**Comment by vinothvino42** [comment]
> @vinothvino42 is this complete? if yes, always close after merging please.

This is not done. We kept this one as a low priority later and worked on other high-priority issues. @vusters Can we do this again?

**Comment by kmahmood74** [comment]
@vusters what's your take, see Vinoth's comment?

**Pull request closed without merging**